### DISCUSSION TITLE
Use vcpkg to build Trinity Core

### DISCUSSION THREAD

**Discussion by chipzz** [TrinityCore/TrinityCore#25829] [Suggestions]
Building TrinityCore is a complex task. Users need to build dependencies manually, and copy the dlls to the build folder to run TrinityCore. Vcpkg addresses both of these issues, so the whole build process may be greatly simplified with vcpkg.

Vcpkg used to be a stand-alone tool, but MicroSoft is working on integration of vcpkg in Visual Studio:
https://devblogs.microsoft.com/cppblog/vcpkg-2020-04-update-and-product-roadmap/#vcpkg-product-roadmap
"5. Vcpkg will be shipped with Visual Studio and Visual Studio Code"

According to https://github.com/microsoft/vcpkg/wiki/Roadmap "Ship vcpkg binary in Visual Studio" was planned for "2020 end of year", but that may have been delayed due to COVID.

More information on vcpkg manifest files can be found at https://devblogs.microsoft.com/cppblog/vcpkg-accelerate-your-team-development-environment-with-binary-caching-and-manifests/ and https://github.com/microsoft/vcpkg/blob/master/docs/specifications/manifests.md

When I tried earlier last year to build the dependencies of TrinityCore with vcpkg, cmake didn't pick up certain dependencies, presumably due to TrinityCore shipping local copies of cmake files with local hacks in cmake/macros/ .

TODO:
- [ ] Code changes:
- - [ ] Add vcpkg.json file;
- - [ ] Add vcpkg_installed to .gitignore (NOTE: not sure if this directory gets created in the projects root or inside the build directory?);
- - [ ] Fix and/ or remove TrinityCore copies of cmake files;
- [ ] TrinityCore wiki changes:
- - [ ] Update TrinityCore wiki with bumped minimal required version of Visual Studio;
- - [ ] Update TrinityCore wiki by removing instructions for dependencies that are built with vcpkg.

**Comment by jackpoz** [comment]
If local cmake macros are causing issues, they could be removed after checking which scenarios become unsupported (and document them and open a ticket to cmake to fix that).

It would be nice if the whole VCPKG was an optional CMake flag that when disabled still keeps the way everything works and when enabled replaces all dependencies with vcpkg.  There might be ways in CMake to exclude local macros when this VCPKG flag is enabled.
